src/main/java/br/unb/cic/analysis/svfa/confluence/SVFAConfluenceAnalysis.java

```diff
@@ -0,0 +1,121 @@
+     * @return a set of confluence conflicts
+     */
+    public Set<ConfluenceConflict> getConfluentConflicts() {
+        return confluentFlows;
+    }
+
+    /**
+     * Executes both source -> base and sink -> base SVFA analysis intersects then populating
+     * the confluentFlows attribute with the results
+     */
+    public void execute() {
+        SVFAAnalysis sourceBaseAnalysis = sourceBaseAnalysis();
+        sourceBaseAnalysis.buildSparseValueFlowGraph();
+        System.out.println(sourceBaseAnalysis.svgToDotModel());
+        Set<List<Node>> sourceBasePaths = sourceBaseAnalysis.findSourceSinkPaths();
+
+        SVFAAnalysis sinkBaseAnalysis = sinkBaseAnalysis();
+        sinkBaseAnalysis.buildSparseValueFlowGraph();
+        System.out.println(sinkBaseAnalysis.svgToDotModel());
+        Set<List<Node>> sinkBasePaths = sinkBaseAnalysis.findSourceSinkPaths();
+
+        confluentFlows = intersectPathsByLastNode(sourceBasePaths, sinkBasePaths);
+    }
+
+    /**
+     * Intersects the list of paths looking for paths that have the same last nodes
+     * also ignores redundant node (that represent different jimple lines but the same Java line)
+     * @param paths1 A set of lists of nodes with at least 2 nodes
+     * @param paths1 A set of lists of nodes with at least 2 nodes
+     * @return A set of confluence conflicts
+     */
+    private Set<ConfluenceConflict> intersectPathsByLastNode(Set<List<Node>> paths1, Set<List<Node>> paths2) {
+        Map<Node, List<Node>> pathEndHash = new HashMap<>();
+
+        for (List<Node> path: paths1) {
+            pathEndHash.put(getLastNode(path), path);
+        }
+
+        Set<ConfluenceConflict> result = new HashSet<>();
+        for (List<Node> path : paths2) {
+            Node lastNode = getLastNode(path);
+            if (pathEndHash.containsKey(lastNode)) {
+                result.add(new ConfluenceConflict(pathEndHash.get(lastNode), path));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * @param path A list of nodes with at least 2 nodes
+     * @return The last node of the list
+     */
+    private Node getLastNode(List<Node> path) {
+        assert path.size() > 1; // assume that all paths have at least one source and one sink
+        return path.get(path.size() - 1);
+    }
+
+    /**
+     * @return A instance of a child class of the SVFAAnalysis class that redefine source and sink as source and base
+     */
+    private SVFAAnalysis sourceBaseAnalysis() {
+        return new SVFAAnalysis(this.cp, this.definition) {
+
+            @Override
+            protected List<Statement> getSourceStatements() {
+                return definition.getSourceStatements();
+            }
+
+            @Override
+            protected boolean isSink(Unit unit) {
+                return !isSource(unit) &&
+                        unit.getJavaSourceStartLineNumber() > 0 && // this is for ignoring lines that are not in any methods body
```

Comment:
@rafaelmotaalves shouldn't we assume here that the soot analysis are only called with lines that correspond to method bodies?

Reply:
Yes, but this line is to guarantee that units outside of the method body are not considered sink, added this because if there is a flow from a source to a method call and the unit corresponding to the method definition is marked as sink, SVFA will return a flow to this method definition. @pauloborba